modules/libpng/1.6.40/patches/add_build_file.patch

```diff
@@ -0,0 +1,62 @@
+--- BUILD.bazel
++++ BUILD.bazel
+@@ -0,0 +1,59 @@
++# Copied from https://github.com/google-deepmind/lab2d/blob/09474439959534b1de3fce3d4a4ed35d565ea230/bazel/png.BUILD 
++# with minor modifications available under Apache 2.0 License
++# SPDX-License-Identifier: Apache-2.0
++
++# Description:
++#   libpng is the official PNG reference library.
++
++genrule(
++    name = "pnglibconf",
++    srcs = ["scripts/pnglibconf.h.prebuilt"],
++    outs = ["pnglibconf.h"],
++    cmd = "cp $(location scripts/pnglibconf.h.prebuilt) $(location pnglibconf.h)",
++)
++
++cc_library(
++    name = "libpng",
```

Comment:
This will result in a library called `liblibpng.so`. if you don't want that, you can use the name `png` and then create an alias named `libpng`.

Reply:
Renamed to png and introduced alias